packages/edit-site/src/hooks/commands/use-page-navigation-commands.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { page as pageIcon } from '@wordpress/icons';
+import { useCommandLoader } from '@wordpress/commands';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as editorStore } from '@wordpress/editor';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const { useHistory } = unlock( routerPrivateApis );
+
+const getPageNavigationCommandLoader = () =>
+	function usePageNavigationCommandLoader( { search } ) {
+		const history = useHistory();
+
+		const { currentPostType, pages, isLoading } = useSelect(
+			( select ) => {
+				const { getCurrentPostType } = select( editorStore );
+				const postType = getCurrentPostType();
+
+				// Only load pages when editing a page and no search query
+				// (the existing search-based loader handles searches)
+				if ( postType !== 'page' || search ) {
+					return {
+						currentPostType: postType,
+						pages: [],
+						isLoading: false,
+					};
+				}
+
+				const query = {
+					per_page: -1,
+					status: 'publish',
```

Comment:
I know this is vibe-coded so apologies if you haven't already done a pass through, but I imagine this should likely include all pages, not just published ones? Or is the goal for users to just be looking at what's published on their site?